### DISCUSSION TITLE
Cannot apply augmentations to RGBA images.

### DISCUSSION THREAD

**Discussion by arelhossan** [TorchIO-project/torchio#577] [Q&A] [answered]
Hello,
I am unsure if this should be created as a bug issue, so I chose the safer option of asking.

In the [Transforms Colab notebook](https://colab.research.google.com/github/fepegar/torchio-notebooks/blob/main/notebooks/Data_preprocessing_and_augmentation_using_TorchIO_a_tutorial.ipynb), it mentions the support for "4D tensors (and therefore also 2D and 3D)". Moreover, in several occasions in the docs (as in the [RandomFlip() documentation](https://torchio.readthedocs.io/transforms/augmentation.html#randomflip)), it mentions 2D tensors. 

However, when trying to augment an RGBA image using this:

```python
img = torch.rand(4, 128, 128)
aug = tio.RandomMotion()
aug(img)
```

It throws this error asking for a 4D tensor:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-24-ee36b2b5fd6a> in <module>()
      1 img = torch.rand(4, 128, 128)
      2 aug = tio.RandomMotion()
----> 3 aug(img)

2 frames
/usr/local/lib/python3.7/dist-packages/torchio/transforms/transform.py in __call__(self, data)
    116             return data
    117         data_parser = DataParser(data, keys=self.include)
--> 118         subject = data_parser.get_subject()
    119         if self.keep is not None:
    120             images_to_keep = {}

/usr/local/lib/python3.7/dist-packages/torchio/transforms/data_parser.py in get_subject(self)
     46             self.is_nib = True
     47         elif isinstance(self.data, (np.ndarray, torch.Tensor)):
---> 48             subject = self._parse_tensor(self.data)
     49             self.is_array = isinstance(self.data, np.ndarray)
     50             self.is_tensor = True

/usr/local/lib/python3.7/dist-packages/torchio/transforms/data_parser.py in _parse_tensor(self, data)
    115                 f' (channels, x, y, z) but it has shape {tuple(data.shape)}'
    116             )
--> 117             raise ValueError(message)
    118         return self._get_subject_from_tensor(data)
    119 

ValueError: The input must be a 4D tensor with dimensions (channels, x, y, z) but it has shape (4, 128, 128)
```
So my question is whether or not I am missing something. How can we feed 2D and 3D tensors to transformations and augmentations?

Thanks in advance!

**Comment by fepegar** [comment, marked as answer]
Hi, @arelhossan.

Maybe this is not very clear in the documentation. 2D images should have `1` as fourth dimension, so the following code should work for you:

```python
import torch
import torchio as tio

img = torch.rand(4, 128, 128, 1)
aug = tio.RandomMotion()
aug(img)
```

Note, though, that `RandomMotion` is a transform to simulate MRI artifacts, so it might be a bit funny to apply it to RGBA images.

**Reply by arelhossan** [reply]
This worked. Thank you very much. 
Thanks for the note as well. I just picked a random augmentation method for testing without going through what it actually does :).

**Reply by fepegar** [reply]
No worries!
